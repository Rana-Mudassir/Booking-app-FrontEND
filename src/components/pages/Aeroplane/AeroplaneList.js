import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchAirplanes } from '../../../store/aeroplane/aeroplanelist';
import { deleteAeroplane } from '../../../store/aeroplane/aeroplane';

const AeroplaneList = () => {
  const dispatch = useDispatch();
  const airplanes = useSelector((state) => state.aeroplaneslist.airplanes);
  const status = useSelector((state) => state.aeroplaneslist.status);

  useEffect(() => {
    dispatch(fetchAirplanes());
  }, [dispatch]);

  const handleDelete = async (id) => {
    const deleteResult = await dispatch(deleteAeroplane(id));
    if (deleteAeroplane.fulfilled.match(deleteResult)) {
      dispatch(fetchAirplanes());
    }
  };

  if (status === 'loading') {
    return <div>Loading...</div>;
  }

  if (status === 'failed') {
    return <div>Error: Unable to fetch airplanes</div>;
  }

  return (
    <div className="AeroList">
      <h2>Aeroplane List</h2>
      <ul className="list">
        {airplanes.map((airplane) => (
          <li className="list_item" key={airplane.id}>
            {airplane.name}
            <button className="loginbtn " type="submit" onClick={() => handleDelete(airplane.id)}>Delete</button>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default AeroplaneList;
