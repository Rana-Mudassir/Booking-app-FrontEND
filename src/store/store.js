import { configureStore } from '@reduxjs/toolkit';
import loginReducer from './authentication/login';
import signupReducer from './authentication/signup';
import aeroplaneReucer from './aeroplane/aeroplane';
import aeroplaneslistReducer from './aeroplaneList/aeroplaneListSlice';
import selectedAirplaneSliceReducer from './aeroplaneList/selectedAirplaneSlice';

const store = configureStore({
  reducer: {

    login: loginReducer,
    signup: signupReducer,
    aeroplane: aeroplaneReucer,
    aeroplaneslist: aeroplaneslistReducer,
    selectedAirplane: selectedAirplaneSliceReducer,
  },
});

export default store;
