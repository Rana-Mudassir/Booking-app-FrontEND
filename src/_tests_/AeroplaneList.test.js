import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../store/store';
import AeroplaneList from '../components/pages/Aeroplane/AeroplaneList';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <AeroplaneList />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
