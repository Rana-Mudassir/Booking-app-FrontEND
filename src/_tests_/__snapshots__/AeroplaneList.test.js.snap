// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="AeroList"
>
  <h2>
    Aeroplane List
  </h2>
  <ul
    className="list"
  />
</div>
`;
